### PR TITLE
Implement a script for generating donations CSV report

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
related to Giveth/giveth-planning/issues/58

Sample file generated by new implemented webservice:

[directDonationsReport.csv.zip](https://github.com/Giveth/feathers-giveth/files/5990996/directDonationsReport.csv.zip)

### CODE REVIEW

File: src/services/directDonationsReport/directDonationsReport.service.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  ];
+};
+const normalizeDonation = donation => {
+  const dappUrl = config.get('dappUrl');
+  const { ownerType, campaign, milestone, tokenAddress } = donation;
+  const token = getTokenByAddress(tokenAddress);
+  const data = {
+    _id: donation._id,
+    usdValue: donation.usdValue,
+    amount: Number(donation.amount).toExponential(),
```
Comment (aminlatifi) [inline src/services/directDonationsReport/directDonationsReport.service.js:32]: Please remove this one

File: src/services/directDonationsReport/directDonationsReport.service.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    amountHumanized: donation.amount / 10 ** 18,
```
Comment (aminlatifi) [inline src/services/directDonationsReport/directDonationsReport.service.js:33]: rename it to amount and use weitoEth function

File: src/services/directDonationsReport/directDonationsReport.service.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  }
+  return data;
+};
+module.exports = async function aggregateDonations() {
+  const app = this;
+  const donationsService = app.service('donations');
+  const donationModel = donationsService.Model;
+
+  const directDonationsReport = async (req, res, _next) => {
+    const campaignDonations = await donationModel.aggregate(createAggregateQuery('campaign'));
```
Comment (aminlatifi) [inline src/services/directDonationsReport/directDonationsReport.service.js:57]: For this https://github.com/Giveth/giveth-planning/issues/58 we don't need aggregate, on the contrary we need list of each of donations, the giver, amount, value, project type (DAC,Campaign,Milestone), link to project, user name (anonymous if not exists)

**Review by aminlatifi — COMMENTED** [review]
It doesn't need to be service,
Query can be simple, just those has homeTxHash and ( ownerType is not user or it has delegateType of dac)

**Comment by mohammadranjbarz** [comment]
@aminlatifi 

is this csv ok?

[https://app.zenhub.com/files/99012473/003f2e6c-1728-4e7d-a7e8-99023db5d3b7/download]

**Comment by aminlatifi** [comment]
> @aminlatifi
> 
> is this csv ok?
> 
> [https://app.zenhub.com/files/99012473/003f2e6c-1728-4e7d-a7e8-99023db5d3b7/download]

Please 
- [ ] Remove _id, amount, ownerTypeId (the link is enough)
- [ ] Rename amountHumanized to amount
- [ ] Add giver name if available
- [ ] Change homeTxHash to `Transaction Hash`, and change all title to human readable ones (like `tokenAddress` to `Token Address` and etc.)
- [ ] Add a column which includes url of tx on etherscan
- [ ]

**Comment by aminlatifi** [comment]
@mohammadranjbarz Some tests are failed

**Comment by mohammadranjbarz** [comment]
> @mohammadranjbarz Some tests are failed, please check in
@aminlatifi done.

**Comment by mohammadranjbarz** [comment]
@aminlatifi 
All changes that you wanted now applyied, you can check the below csv

[https://app.zenhub.com/files/99012473/d8f16803-32f5-443f-b789-f1b0d4a910bf/download]

File: src/services/directDonationsReport/directDonationsReport.service.js (hunk 1 above)
Comment (mohammadranjbarz) [reply src/services/directDonationsReport/directDonationsReport.service.js]: done

File: src/services/directDonationsReport/directDonationsReport.service.js (hunk 2 above)
Comment (mohammadranjbarz) [reply src/services/directDonationsReport/directDonationsReport.service.js]: done

**Comment by mohammadranjbarz** [comment]
[directDonationsReport.csv.zip](https://github.com/Giveth/feathers-giveth/files/6057409/directDonationsReport.csv.zip)

@aminlatifi  You can check this csv now

**Comment by aminlatifi** [comment]
> [directDonationsReport.csv.zip](https://github.com/Giveth/feathers-giveth/files/6057409/directDonationsReport.csv.zip)
> 
> @aminlatifi You can check this csv now

It seems OK @mohammadranjbarz , Thanks

**Comment by aminlatifi** [comment]
- [ ] Remove pledgeId
- [ ] Make timezone UTC+0 (not exported machine)

**Comment by mohammadranjbarz** [comment]
@aminlatifi done

**Comment by GriffGreen** [comment]
@mohammadranjbarz This is awesome!! 

Can I have one more column that displays the Campaign name (if there is one) If it's a DAC its fine to be blank... but if its a milestone, we want the campaign name still, not the milestone name.

**Comment by aminlatifi** [comment]
> @mohammadranjbarz This is awesome!!
> 
> Can I have one more column that displays the Campaign name (if there is one) If it's a DAC its fine to be blank... but if its a milestone, we want the campaign name still, not the milestone name.

@mohammadranjbarz

**Comment by mohammadranjbarz** [comment]
> @mohammadranjbarz This is awesome!!
> 
> Can I have one more column that displays the Campaign name (if there is one) If it's a DAC its fine to be blank... but if its a milestone, we want the campaign name still, not the milestone name.

@GriffGreen 
Sure I'll work on it

**Comment by mohammadranjbarz** [comment]
[directDonationsReport.csv.zip](https://github.com/Giveth/feathers-giveth/files/6061827/directDonationsReport.csv.zip)


@GriffGreen I added the Campaign name column , you can check this new one

**Pull request merged**